### PR TITLE
:ship: Adds beginning of privacy notice page

### PR DESCRIPTION
**Pull request by S0AndS0**

### CODE REVIEW

File: public/privacy.html [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+          crossorigin="anonymous">
+  </head>
+
+  <body>
+    <strong>
+      Currently the technologies utilized by this site may cause submitted graph data to be saved within server logs.
+    </strong>
+
+    <p>
+      By utilizing features of this site you're implying that you agree that you have read and unserstood this notice.
```
Comment (oferb) [inline public/privacy.html:19]: `unserstood -> understood`

File: public/privacy.html [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    </p>
+
+    <p>
+      Until this notice is removed please do <strong>not</strong> use graph-to-braille or graph-to-sound features of this site for any data that is confidential or private.
```
Comment (oferb) [inline public/privacy.html:23]: I'd remove "graph-to-braille or graph-to-sound features of ". Just:

"use this site for any data that is confidential or private."

File: public/privacy.html [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+    <link rel="stylesheet"
+          href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/css/bootstrap.min.css"
+          integrity="sha384-Gn5384xqQ1aoWXA+058RXPxPg6fy4IWvTNh0E263XmFcJlSAwiGgFAW/dAiS6JXm"
+          crossorigin="anonymous">
+  </head>
+
+  <body>
+    <strong>
+      Currently the technologies utilized by this site may cause submitted graph data to be saved within server logs.
```
Comment (oferb) [inline public/privacy.html:15]: Currently the technologies utilized by this site place graph data in the url, which can be sent to our server in some cases. The reason we do this:
1. It simplifies the development
2. It makes it easy to _share_ data (you just send the url)
3. It makes it easy to _store_ data (you just save the url)

Here's an example:
https://sensoryinterface.com/view/index.html?data=1%092%093
"1%092%093" is the encoded data, which, when decoded becomes "1,2,3".

Once this url reaches our server it maybe be logged by our hosting provider, which is currently Google.
We don't store this data ourselves, but, we don't have a way in which you can verify it.

File: public/privacy.html (hunk 2 above)
Comment (oferb) [inline public/privacy.html:23]: "please do <strong>not</strong>"

I think it's up to the user, once they've understood, to make their choice.
I'd rephrase to:
"Until this notice is removed please be aware of the above if using this site for any data that is confidential or private."

File: public/privacy.html (hunk 1 above)
Comment (S0AndS0) [reply public/privacy.html:19]: Heh, indeed this is a live example for why I prefer to write MarkDown files and have automation transpile to HTML, because spell-checkers are usable within the former and overly sensitive in the latter.

I'll be correcting the spelling within the next push.

File: public/privacy.html (hunk 2 above)
Comment (S0AndS0) [reply public/privacy.html]: Thanks, corrections shall be within the next push.

File: public/privacy.html (hunk 3 above)
Comment (S0AndS0) [reply public/privacy.html:15]: Considering anyone can Fork this project and host it elsewhere, I think that keeping this generalized will be simpler and less likely to be incorrect.

File: public/privacy.html (hunk 2 above)
Comment (S0AndS0) [reply public/privacy.html]: I'm going to have to disagree with ya on that.

If this project make it clear that clients shouldn't use it for private or confidential data, and the do so anyways, then it's less likely that authors/maintainers can be held liable. Where as wording things as "it's up to clients" allows too much _wiggle room_ for who could be held responsible for leaks.

File: public/privacy.html (hunk 3 above)
Comment (oferb) [reply public/privacy.html:15]: If someone forks and hosts then its their responsibility to modify this file.

File: public/privacy.html (hunk 2 above)
Comment (oferb) [reply public/privacy.html]: IANAL and I'm sure there's more inviting ways to write this, but I don't think too many people read the privacy notice, so we'll likely not scare too many people off.

File: public/privacy.html (hunk 3 above)
Comment (S0AndS0) [reply public/privacy.html:15]: I remember that keeping things simple as much as possible is preferred. And considering that the requirement of modifying this specific file is not stated within the ReadMe file, or anywhere else other than this thread, I very much doubt that it can be expected for anyone to know that modifying this file prior to hosting elsewhere is needed.

File: public/privacy.html (hunk 2 above)
Comment (S0AndS0) [reply public/privacy.html]: It's not about scaring but instead about informing people.

Comment (oferb) [reply public/privacy.html]: I sometimes get scared by terms of service, especially when unclear and/or alarming. I'm sure there are others like me.

File: public/privacy.html (hunk 3 above)
Comment (oferb) [reply public/privacy.html:15]: Keeping things simple for people contributing to _this_ repo, not a fork :)

File: public/privacy.html (hunk 2 above)
Comment (S0AndS0) [reply public/privacy.html]: If you've a better way of phrasing things then feel free to open a PR so that attribution is maintained.

The notice is clear, and no one has control over another persons emotional state.

What I find alarming is that at this time anyone from the EU, or other privacy protecting regions, could _cry foul_ and litigate this project into oblivion.

**Comment by S0AndS0** [comment]
Found a template for writing a GDPR compliant privacy notice...

https://gdpr.eu/privacy-notice/

... I'll spend a bit of time adding related sections. however, some will likely need to be filled in by Organization owners, **and** even after beware that there are other regions that have their own laws that need to be accounted for; ie I heard roomers that the UK has their own regulations on client submitted data based on (but not identical to) the GDPR.

File: public/privacy.html (hunk 2 above)
Comment (oferb) [reply public/privacy.html]: There is nothing to litigate against because no data is actually stored.

Comment (oferb) [reply public/privacy.html]: How about:

"Until this notice is removed please avoid using this site for any data that is confidential or private."

**Comment by oferb** [comment]
You wrote:

"Sensory Interface collects the following data: Information submitted via URL Search Params"

We don't collect this data. In fact, we don't collect any data.

**Comment by S0AndS0** [comment]
That line is adapted from the [GDPR Template](https://gdpr.eu/privacy-notice/), and it's written that way because, servers collect **and** log Search Params by default.


Think about it as a chain of custody;


- the client provides `sensoryinterface.com` with information

- that information is logged by _Hosing Provider "G"_


Regardless of if anyone working on the Sensory Interface project is able to access that information, both `sensoryinterface.com` **and** _Hosing Provider "G"_ have indeed collected information from the client.


___


I'll encourage searching for terms such as "do I need to be GDPR compliant", here's one helpful resulting site...


- [TrueVault -- Do I need to be GDPR compliant?](https://www.truevault.com/blog/do-i-need-to-be-gdpr-compliant.html)


... here are some of the note worthy points...


> ... Moreover, the fines for breaching the Regulation are harsh, going up to €20,000,000 or 4% of your global turnover (whichever is higher)...
>
> ... you will be affected if you are a data controller or data processor and any of the following apply
>
>
> - you are established in the EU (or somewhere else subject to EU law), or
> - you offer goods or services to individuals in the EU, or
> - you monitor the behavior of individuals in the EU.


------


... and here's some highlights on what entities are considered to be a "data controller" and a "data processor"...


- [TrueVault -- What is the difference between a data controller and a data processor?](https://www.truevault.com/learn/what-is-the-difference-between-a-data-controller-and-a-data-processor)

> ... In practice, that means that a data controller is responsible for determining how and why data is going to be used by an organization...
>
> ... A data processor does not own any of the data they process and are not in control of it...

**Comment by oferb** [comment]
You convinced me :)

Let's do GDPR in a different PR then, ok? This one will focus on `public/privacy.html`.

This thread is still open:
https://github.com/hasadna/sensory-interface/pull/105#discussion_r423962690

File: public/privacy.html (hunk 2 above)
Comment (S0AndS0) [reply public/privacy.html]: While some may still choose to misunderstand the way it's currently phrased, I prefer direct wording that mitigates the chances of misinterpretation; ie. allow some even a little room to _wiggle_, and liability could be passed to this project for their decisions.

File: public/privacy.html (hunk 3 above)
Comment (oferb) [inline public/privacy.html:15]: > within server logs

Maybe add that we use Google's Firebase Hosting and mention that we don't log data, but Google might?

Comment (S0AndS0) [reply public/privacy.html:15]: How about something like?...

> Sensory Interface asserts that we do not have access to server logs, however, hosting providers (such as Google's Firebase) may log server interactions and/or archive server logs.

Comment (oferb) [reply public/privacy.html:15]: Sounds great!
